Sources_v3/StreamChat/Config/Token.swift

```diff
@@ -1,40 +1,41 @@
 //
-// Copyright © 2020 Stream.io Inc. All rights reserved.
+// Copyright © 2021 Stream.io Inc. All rights reserved.
 //
 
 import Foundation
 
-/// A token is used to authenticate a user.
-public typealias Token = String
+enum TokenError: Error {
+    case missingUserId
+}
 
-/// A token provider is a function in which you send a request to your own backend to get a Stream Chat API token.
-///
-/// Set your custom `TokenProvider` in `ChatClientConfig` when creating a `ChatClient` instance. `ChatClient` will use it whenever
-/// it needs to get a new token for the given user.
-///
-public typealias TokenProvider = (_ apiKey: APIKey, _ userId: UserId, _ completion: @escaping (Token?) -> Void) -> Void
+public struct Token: Decodable, Equatable {
```

Comment:
Are we going to handle expiring tokens in a different PR?

Reply:
@b-onc there's even a JIRA ticket for that already :)

Reply:
Would be nice it have it done inside this PR.

@VojtaStavik wdyt?